models/migrations/v126.go

```diff
@@ -10,40 +10,16 @@ import (
 )
 
 func fixTopicRepositoryCount(x *xorm.Engine) error {
-	type Topic struct {
-		ID        int64
-		RepoCount int
-	}
-
 	_, err := x.Exec(builder.Delete(builder.NotIn("`repo_id`", builder.Select("`id`").From("`repository`"))).From("`repo_topic`"))
 	if err != nil {
 		return err
 	}
 
-	var last int
-	const batchSize = 50
-	for {
-		var results = make([]Topic, 0, batchSize)
-		err := x.OrderBy("id").
-			Limit(batchSize, last).
-			Find(&results)
-		if err != nil {
-			return err
-		}
-		if len(results) == 0 {
-			break
-		}
-		last += len(results)
-
-		for _, res := range results {
-			_, err := x.ID(res.ID).Cols("repo_count").
-				SetExpr("repo_count", builder.Select("count(*)").From("repo_topic").Where(
-					builder.Eq{"topic_id": res.ID},
-				)).Update(res)
-			if err != nil {
-				return err
-			}
-		}
-	}
-	return nil
+	_, err = x.Exec(builder.Update(
+		builder.Eq{
+			"`repo_count`": builder.Select("count(*)").From("`repo_topic`").Where(builder.Eq{
+				"`repo_topic`.`topic_id`": builder.Expr("`topic`.`id`"),
+			}),
+		}).From("`topic`").Where(builder.Eq{"'1'": "1"}))
```

Comment:
Interesting!

Reply:
Such cunning evilness

Reply:
I just wanted to see if it can be done :laughing: